File: BE/src/main/java/com/example/hotel/model/entity/RoomType.java
package com.example.hotel.model.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Builder
@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name="room_type")
public class RoomType {

    @Id
    @GeneratedValue(strategy= GenerationType.IDENTITY)
    private Long id;

    private String typeName;

    private BigDecimal price;

    private Double size;

    private Double area;

    private String layout;

    private String extraService;

    @JsonManagedReference
    @OneToMany(fetch = FetchType.EAGER, mappedBy = "roomType", cascade = CascadeType.ALL)
    private List<Image> imageList;

    @Lob
    private String shortDescription;

    @Lob
    private String description;

    @CreationTimestamp
    private LocalDateTime createdTime;

    @UpdateTimestamp
    private LocalDateTime updatedTime;
}

File: FE/src/app/core/model/room-type.ts
export class roomType {
  id: number;
  typeName: string;
  price: number;
  description: string;
  shortDescription: string;
  area: number;
  extraService: string;
  layout: string;
  size: number;

  constructor() {
    this.id = 0
    this.typeName = "";
    this.price = 0;
    this.description = "";
    this.shortDescription = "";
    this.area = 0;
    this.size = 0;
    this.extraService = "";
    this.layout = "";
  }
}

export class imgResponseList {
  imgEncodeString: string;
  fileCode: string;

  constructor() {
    this.imgEncodeString = '';
    this.fileCode = '';
  }
}
